### PR TITLE
(PCP-209) Unvendor and rely on installed leatherman

### PR DESCRIPTION
**Pull request by parisiale**
Removing leatherman's git submodule. Updating CMake, Appveyor, and
Travis configuration scripts. Updating leatherman's dependency on
README.

### CODE REVIEW

**Comment by mruzicka** [comment]
LGTM.
@richardc Can you please review this too?

File: appveyor.yml [hunk 1]
```diff
@@ -14,12 +17,14 @@ install:
   - SET OPENSSL_ROOT_DIR=C:\tools
   - SET PATH=C:\tools\bin;%PATH%
 
+  - ps: wget 'https://s3.amazonaws.com/kylo-pl-bucket/curl-7.42.1-x86_64_mingw-w64_4.8.3_win32_seh.7z' -OutFile "$pwd\curl-7.42.1-x86_64_mingw-w64_4.8.3_win32_seh.7z"
+  - ps: 7z.exe x "curl-7.42.1-x86_64_mingw-w64_4.8.3_win32_seh.7z" -oC:\tools | FIND /V "ing "
```
Comment (richardc) [inline appveyor.yml]: This seems new, and not something we need.

Comment (parisiale) [inline appveyor.yml]: Yeah, it's needed by the leatherman 0.3.5 package we're using; its `leatherman/lib/cmake/leatherman/LeathermanConfig.cmake` includes ```
leatherman_component(curl)``` which was breaking the build. See https://ci.appveyor.com/project/puppetlabs/cpp-pcp-client/build/1.0.116.

I was planning to get this in and then make a ticket to fix this useless dependency.
Do you have an alternative to fix? Perahps, @branan or @MikaelSmith, I'm doing something wrong.

Comment (richardc) [inline appveyor.yml]: Oh, that's a binary artifact of leatherman? I guess we can live with it for now, though we could consider using a source build of leatherman as we do under travis?

Comment (MikaelSmith) [inline appveyor.yml]: That's a mistake in Leatherman. This PR does use a source build of Leatherman on Windows, but it's pulling in a required dependency on libcurl even though you're not requesting the Leatherman.curl component. I'll file a ticket to fix it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -37,38 +36,29 @@ if ("${INSTALL_IS_SYSTEM_DIR}" STREQUAL "-1")
...
+find_package(Leatherman REQUIRED
+  COMPONENTS locale nowide catch logging rapidjson json_container util)
+find_package(Boost 1.54 REQUIRED
+  COMPONENTS filesystem system date_time thread log regex random)
+find_package(OpenSSL REQUIRED)
 
 # Leatherman it up
 include(options)
+include(cflags)
 option(BOOST_STATIC "Use Boost's static libraries" OFF)
```
Comment (MikaelSmith) [inline CMakeLists.txt]: This shouldn't be needed anymore, it's in `include(options)`.

**Comment by MikaelSmith** [comment]
Looks good, just a minor nit.

**Comment by parisiale** [comment]
Thanks. I removed the `BOOST_STATIC` def.

**Comment by parisiale** [comment]
Just for reference, I've created PCP-246 to follow LTH-76.

**Pull request merged**